File: packages/cli/test/fixtures/ranking-corpora/csharp/Logger.cs
namespace RankingCSharp;

/// <summary>
/// Shared logging helper -- the hub type every handler in this fixture
/// corpus references. Deliberately same-namespace with no `using` needed
/// anywhere in this corpus, which is why the reverse-dependency signal here
/// comes entirely from the C# type-reference recovery tier (#930/#943)
/// rather than the regular import graph.
/// </summary>
public static class Logger
{
    public static void LogInfo(string message)
    {
        Console.WriteLine($"[info] {message}");
    }

    public static void LogError(string context, Exception err)
    {
        Console.WriteLine($"[error] {context}: {err.Message}");
    }
}

File: packages/cli/test/fixtures/ranking-corpora/python/rankingpy/util/logger.py
"""Shared logging helpers -- the hub module every handler in this fixture
corpus imports."""


def log_info(message: str) -> None:
    """Write an informational message, prefixed for easy identification."""
    print(f"[info] {message}")


def log_error(context: str, err: Exception) -> None:
    """Write an error message, wrapping err with context."""
    print(f"[error] {context}: {err}")
